contrib/mypy/src/python/pants/contrib/mypy/tasks/mypy_task.py

```diff
@@ -20,25 +20,32 @@
 from pex.pex import PEX
 from pex.pex_info import PexInfo
 from pants.task.lint_task_mixin import LintTaskMixin
+from typing import List
 
 
 class MypyTaskError(TaskError):
-  """Indicates a TaskError from a failing MyPy run"""
+  """Indicates a TaskError from a failing MyPy run."""
 
 class MypyTask(LintTaskMixin, ResolveRequirementsTaskBase):
   """Invoke the mypy static type analyzer for Python.
 
   Mypy lint task filters out target_roots that are not properly tagged according to
-  --whitelisted-tag-name, and executes my py on targets in context from whitelisted target
-  roots. Next, if any transitive targets from the filtered roots are not whitelisted, a Mypy
-  TaskError will be thrown.
+  --whitelisted-tag-name (defaults to None, and no filtering occurs if this option is 'None'),
+  and executes MyPy on targets in context from whitelisted target roots.
+  Next, if any transitive targets from the filtered roots are not whitelisted, a Warning.
+  will be thrown.
 
   'In context' meaning in the sub-graph where a whitelisted target is the root
   """
 
   _MYPY_COMPATIBLE_INTERPETER_CONSTRAINT = '>=3.5'
   _PYTHON_SOURCE_EXTENSION = '.py'
 
+  deprecated_options_scope = 'mypy'
+  deprecated_options_scope_removal_version = '1.20.0.dev2'
+
+  WARNING_MESSAGE = None
```

Comment:
added this field for testing purposes, didn't seem worth spending a lot of time trying to figure out a way to keep track of logging/warning messages, so I used this workaround for now.

Reply:
You can hardcode the value here. 
```
WARNING_MESSAGE = "[WARNING]: All targets in context should be whitelisted for mypy to run"
```